src/main/java/net/countercraft/movecraft/combat/movecraftcombat/directors/CannonDirectorManager.java

```diff
@@ -171,6 +177,25 @@ private void processDirectors() {
 
     }
 
+    private Craft getDirectingCraft(TNTPrimed tnt)
+    {
+        List<MetadataValue> meta = tnt.getMetadata("MCC-Sender");
+
+        if(Config.EnableTNTTracking && !meta.isEmpty()) {
```

Comment:
To simplify the readability, I would format it as following:
```
if not enabled,
  return null

*get meta*
if meta is empty,
  return null

*get sender*
if sender null or offline,
  return null

*get craft*
if null craft or sinking,
  return null
return craft
```

And then if the directing craft is null, search via the old method.

Reply:
This will reduce the lag impact on servers by only getting the metadata if the feature is enabled, only getting the player from the UUID if the meta isn't empty, and only getting a craft if the player is still online and valid.